pygsti/circuits/circuit.py

```diff
@@ -4681,12 +4702,35 @@ class SeparatePOVMCircuit(object):
     """
     def __init__(self, circuit_without_povm, povm_label, effect_labels):
         self.circuit_without_povm = circuit_without_povm
-        self.povm_label = povm_label
-        self.effect_labels = effect_labels
+        self._povm_label = povm_label
+        self._effect_labels = effect_labels
+        self._full_effect_labels = tuple([(self.povm_label + "_" + el) for el in self._effect_labels])
 
     @property
     def full_effect_labels(self):
-        return [(self.povm_label + "_" + el) for el in self.effect_labels]
+        return self._full_effect_labels
+    
+    @property 
+    def effect_labels(self):
+        return self._effect_labels
+    
+    @property 
+    def povm_label(self):
+        return self._povm_label
+
+    @effect_labels.setter
```

Comment:
Should setters have static checks?

Reply:
SeparatePOVMCircuit doesn't presently have a static flag. I'd have to take a closer look at how these are used in practice elsewhere in the code (i.e. whether other parts of the code are interacting with these attributes. They are of course leading underscored, so maybe not? My main motivation for adding the setters was that the value of _full_effect_labels is derived from the values of _povm_label and _effect_labels, so I wanted to make it slightly more difficult to have a situation where they could get out of sync now that _full_effect_labels is being cached.

Reply:
Didn't realize this was in SeparatePOVMCircuit, my comment is less relevant in this case and I don't think adding static to a new class is needed for this PR :)